cxx/clean_relation.hh

```diff
@@ -96,6 +96,7 @@ class CleanRelation : public Relation<T> {
     }
     T_items z = get_cluster_assignment(items);
     if (!clusters.contains(z)) {
+      assert(prng != nullptr);
```

Comment:
Passing a nullptr for prng when we want to make sure the prng isn't used is a new usage of this method introduced in this PR, and debugging this assert is easier than a crash because something tried to use a nullptr.